tests/test_namespace_isolation.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+import random
+import string
+import pytest
+
+import jubilant
+import temporalio.client
+
+from .helpers import create_workflow
+
+logger = logging.getLogger(__name__)
+
+
+async def test_python_workflow(
+    python_namespace: str,
+    python_task_queue: str,
+    python_workflow_classname: str,
+    juju_server_model: jubilant.Juju,
+):
+    """Test to ensure that the python workflow can be executed"""
+    logger.info("Creating python workflow manually")
+
+    random_keyword = "".join(random.choices(string.ascii_letters + string.digits, k=10))
+
+    workflow_id = create_workflow(
+        juju_server_model,
+        python_namespace,
+        python_task_queue,
+        python_workflow_classname,
+        random_keyword,
+    )
+
+    logger.info("Ensuring only workflow exists")
+
+    temporal_server_unit_ip = juju_server_model.status().apps["temporal-k8s"].address
+    assert temporal_server_unit_ip != "", "Empty address for temporal-k8s application"
+
+    client = await temporalio.client.Client.connect(
+        f"{temporal_server_unit_ip}:7233", namespace=python_namespace
+    )
+
+    python_workflow_ids = []
+    async for workflow in client.list_workflows():
+        python_workflow_ids.append(workflow.id)
+
+    assert workflow_id in python_workflow_ids, "Unable to find created workflow in all workflows"
+
+    logger.info("Waiting until workflow completes")
+
+    workflow_handle = client.get_workflow_handle(workflow_id)
+    workflow_result = await workflow_handle.result()
+    assert workflow_result == f"Hello world to {random_keyword} in python!", (
+        "Unexpected result in activity of created workflow"
+    )
+
+
+async def test_go_workflow(
```

Comment:
I see a lot of shared operations with the previous test case. Thinking ahead (when we start supporting more SDKs), should we abstract this to a method that can be called from within each test SDK case?

Reply:
Added task https://github.com/canonical/charmed-temporal-uats/issues/5